test/suite/helper_tiered.py

```diff
@@ -84,53 +89,79 @@ def get_check(storage_source, tc, base, n):
+    prefix += datetime.datetime.now().strftime('%Y-%m-%d-%H-%M') + '--' + random_prefix +'/'
+    # Group all scenarios from the same test under the same test name.
+    prefix += test_name + '/'
+
+    # Generate a random number to differentiate object files for tests that use multiple bucket
+    # prefixes. It is important to generate unique prefixes for different tests in the same class,
     # so that the database namespace do not collide.
-    # 0th element on the stack is the current function. 1st element is the calling function.
-    if not test_name:
-        test_name = inspect.stack()[1][3]
-    prefix += test_name + '--'
+    # Range up to int32_max, matches that of C++'s std::default_random_engine
+    prefix += str(random.randrange(1, 2147483646)) + '--'
+
     return prefix
 
-# Storage sources
-tiered_storage_sources = [
-    ('dirstore', dict(is_tiered = True,
-        is_local_storage = True,
-        auth_token = get_auth_token('dir_store'),
-        bucket = get_bucket_name('dir_store', 0),
-        bucket1 = get_bucket_name('dir_store', 1),
-        bucket_prefix = "pfx_",
-        bucket_prefix1 = "pfx1_",
-        bucket_prefix2 = 'pfx2_',
-        num_ops=100,
-        ss_name = 'dir_store')),
-    ('s3', dict(is_tiered = True,
-        is_local_storage = False,
-        auth_token = get_auth_token('s3_store'),
-        bucket = get_bucket_name('s3_store', 0),
-        bucket1 = get_bucket_name('s3_store', 1),
-        bucket_prefix = generate_s3_prefix(),
-        bucket_prefix1 = generate_s3_prefix(),
-        bucket_prefix2 = generate_s3_prefix(),
-        num_ops=20,
-        ss_name = 's3_store')),
-    ('non_tiered', dict(is_tiered = False)),            
-]
-
-# Sublist to use for the tiered test scenarios as last item on list is not a scenario.  
-storage_sources = tiered_storage_sources[:2]
+def gen_storage_sources(random_prefix='', test_name=''):
+    tiered_storage_sources = [
+        ('dirstore', dict(is_tiered = True,
+            is_local_storage = True,
+            auth_token = get_auth_token('dir_store'),
+            bucket = get_bucket_name('dir_store', 0),
+            bucket1 = get_bucket_name('dir_store', 1),
+            bucket_prefix = "pfx_",
+            bucket_prefix1 = "pfx1_",
+            bucket_prefix2 = 'pfx2_',
+            num_ops=100,
+            ss_name = 'dir_store')),
+        ('s3', dict(is_tiered = True,
+            is_local_storage = False,
+            auth_token = get_auth_token('s3_store'),
+            bucket = get_bucket_name('s3_store', 0),
+            bucket1 = get_bucket_name('s3_store', 1),
+            bucket_prefix = generate_s3_prefix(random_prefix, test_name),
+            bucket_prefix1 = generate_s3_prefix(random_prefix, test_name),
+            bucket_prefix2 = generate_s3_prefix(random_prefix, test_name),
+            num_ops=20,
+            ss_name = 's3_store')),
+        ('non_tiered', dict(is_tiered = False)),
```

Comment:
This item doesn't seem to get used anymore. If is is no longer required,  we can consider removing this.

Reply:
Good point, I think other tests might still use this so I've added an argument to `gen_storage_sources` so that it returns the whole scenario list unless only the tiered scenarios are required.